### PR TITLE
Always set RestoreUseStaticGraphEvaluation to true

### PR DESCRIPTION
**Pull request by sharwell**
Opening so we can discuss why this would or would not be desirable.

/cc @tmat @jmarolf

### CODE REVIEW

**Comment by jmarolf** [comment]
@sharwell there is no reason to not do this. The VS version we were building against in the past was not always compatible with this option. However, this change should be made in build.ps1 [here](https://github.com/dotnet/roslyn/blob/7fea27afc959115841d671c561afe8d679d8db7c/eng/build.ps1#L270)

**Comment by Youssef1313** [comment]
@sharwell https://github.com/NuGet/Home/issues/9803 tracks enabling it by default. Probably it was disabled by default as it's experimental feature.

**Comment by jmarolf** [comment]
1. I forget that the VS restore is different :( my main situation was trying to improve CI times
2. and 3. ¯\_(ツ)_/¯ up to nuget team

**Comment by sharwell** [comment]
@nkolev92 any reason for us to not merge this?

**Review by jmarolf — APPROVED** [review]

**Comment by nkolev92** [comment]
Nope, as long as you do the equivalency test mentioned in the body of https://github.com/NuGet/Home/issues/9803 and it passes, you are all good.

**Comment by sharwell** [comment]
Verified that equivalency test passed.

File: eng/targets/Settings.props [hunk 1]
```diff
@@ -12,7 +12,9 @@
 
     <CommonExtensionInstallationRoot>CommonExtensions</CommonExtensionInstallationRoot>
     <LanguageServicesExtensionInstallationFolder>Microsoft\ManagedLanguages\VBCSharp\LanguageServices</LanguageServicesExtensionInstallationFolder>
-    
+
```
Comment (jaredpar) [inline eng/targets/Settings.props:15]: Can we avoid blank lines before after properties in property blocks? Unless there is a comment prefer we just group them together.

**Comment by jaredpar** [comment]
Why didn't we delete this when merging this change? 

https://github.com/dotnet/roslyn/blob/main/eng/build.ps1#L270

**Comment by sharwell** [comment]
> Why didn't we delete this when merging this change?

The line in build.ps1 applies to pseudo-projects with broader scope than what gets applied for restore operations that occur within Visual Studio.

**Pull request merged**